src/generic/FreeAssociativeAlgebra.jl

```diff
@@ -633,6 +637,42 @@ end
 #
 ###############################################################################
 
+function *(a::FreeAssociativeAlgebraElem, n::Union{Integer, Rational, AbstractFloat})
+    z = zero(a)
+    fit!(z, length(a))
+    j = 1
+    for i = 1:length(a)
+        c = a.coeffs[i]*n
+        if !iszero(c)
+            z.coeffs[j] = c
+            z.exps[j] = a.exps[i]
+            j += 1
+        end
+    end
+    z.length = j - 1
+    return z
+end
+
+function *(a::FreeAssociativeAlgebraElem{T}, n::T) where {T <: RingElem}
+    z = zero(a)
+    fit!(z, length(a))
+    j = 1
+    for i = 1:length(a)
+        c = a.coeffs[i]*n
+        if !iszero(c)
+            z.coeffs[j] = c
+            z.exps[j] = a.exps[i]
+            j += 1
+        end
+    end
+    z.length = j - 1
+    return z
+end
```

Comment:
How about reducing code duplication and increasing flexibility by doing this instead:
```suggestion
function mul!(z::T, a::T, n::RingElement) where {T <: FreeAssociativeAlgebraElem}
    fit!(z, length(a))
    j = 1
    for i = 1:length(a)
        c = a.coeffs[i]*n
        if !iszero(c)
            z.coeffs[j] = c
            z.exps[j] = a.exps[i]
            j += 1
        end
    end
    z.length = j - 1
    return z
end

*(a::FreeAssociativeAlgebraElem, n:: Union{Integer, Rational, AbstractFloat}) = mul!(z, a, n)

*(a::FreeAssociativeAlgebraElem{T}, n::T) where {T <: RingElem} = mul!(z, a, n)
```

Reply:
The signature for `mul!` here is too wide, as this would capture `mul!(::T, ::T, ::T) where T <: FreeAssociativeAlgebraElem` as well. But I added something similar, with a restricted signature